python/ray/tests/test_state_api.py

```diff
@@ -2068,6 +2069,231 @@ def ready(self):
     assert "actor_id" in result.output
 
 
+def try_state_query(api_func, q):
+    """Utility functions for rate limit related e2e tests below"""
+    try:
+        api_func()
```

Comment:
Actually this will not affect the running time I think. 
In the test `test_state_api_rate_limit_with_failure`  which I test failures of the list_X apis by simply killing the processes, the delay of 10 sec on the grpc was never hit. 

In the test `test_state_api_server_enforce_concurrent_http_requests`, the delay is 2s so all calls should finish within 2 seconds, and the test does wait for the calls to return.